apps-rendering/src/components/LiveBlocks/index.tsx

```diff
@@ -44,8 +61,12 @@ const LiveBlocks: FC<LiveBlocksProps> = ({ blocks, format }) => {
 					blockTitle={block.title}
 					blockFirstPublished={Number(block.firstPublished)}
 					blockId={block.id}
-					// TODO pass this value in when available
+					// This is false because it's only true when it's
+					// used for the actual pinned post on top of the page
 					isPinnedPost={false}
+					isOriginalPinnedPost={
+						block.isPinned && block.id === pinnedPostId
```

Comment:
Good question. From my testing with the CAPI data, I don't think this should ever happen because as soon as a post is pinned, the pinned attribute in previous pinned post would be set to false. 

But I noticed that DCR is doing this check so I thought there's no harm to do so in here too. Just in case if get 2 blocks that have their pinned attribute set to true. 

But I'm interested to hear what you or others think :)

Reply:
A possible alternative implementation that removes the need to set `pinnedPostId` to `''`:

```suggestion
						pipe(
							pinnedPost,
							map(pinned => block.isPinned && block.id === pinned.id),
							withDefault(false),
						)
```

Also just wondering: if the current block id is equal to the pinned block id, do you also need the check for `isPinned`? You've already used `isPinned` to get the pinned block id in the first place I think?

Reply:
You're absolutely right, we can only rely on the block id here 👍

Reply:
The pipe option looks great. I just had to explicitly add the type for `withDefault` otherwise the storybook build would have failed with the following error (the error is so strange 🤔 ):
```
      TS2345: Argument of type '(optA: Option<LiveBlock>) => Option<boolean>' is not assignable to parameter of type '(_a: Option<LiveBlock>) => Option<false>'.
  Type 'Option<boolean>' is not assignable to type 'Option<false>'.
    Type 'Some<boolean>' is not assignable to type 'Option<false>'.
      Type 'Some<boolean>' is not assignable to type 'Some<false>'.
        Type 'boolean' is not assignable to type 'false'.
```

But the build was successful with this:
```
pipe(
  pinnedPost,
  map((pinned) => block.id === pinned.id),
  withDefault<boolean>(false),
)
```

Reply:
I think the type inference has probably narrowed things too far here. It's inferred the type as the `false` when it's actually `boolean`, which is `true | false`. It's then become unhappy because it can't assign the wider type `boolean` to `false`.

I don't really understand why this would only come up in the storybook build though, given that the main build doesn't seem to have this problem...